### PR TITLE
feat: add Seurat step function

### PR DESCRIPTION
**Pull request by ebezzi**
### Reviewers
**Functional:** 
@metakuni 

**Readability:** 

---

## Changes
- Adds the Seurat step function, as already defined in [sinle-cell-infra](https://github.com/chanzuckerberg/single-cell-infra/pull/461)

### CODE REVIEW

File: .happy/terraform/modules/sfn/main.tf [hunk 1]
```diff
@@ -195,6 +195,69 @@ resource "aws_sfn_state_machine" "state_machine" {
...
+            },
+            {
+              "Name": "STEP_NAME",
+              "Value": "seurat"
+            }
+          ]
+        }
+      },
+      "TimeoutSeconds": 36000,
+      "Retry": [
```
Comment (metakuni) [inline .happy/terraform/modules/sfn/main.tf:229]: We probably don't need retries on conversion failures.

File: .happy/terraform/modules/sfn/main.tf [hunk 2]
```diff
@@ -195,6 +195,69 @@ resource "aws_sfn_state_machine" "state_machine" {
...
+{
+  "StartAt": "Seurat",
+  "States": {
+    "Seurat": {
+      "Type": "Task",
+      "End": true,
+      "Resource": "arn:aws:states:::batch:submitJob.sync",
+      "Parameters": {
+        "JobDefinition": "${var.job_definition_arn}",
+        "JobName": "cxg",
```
Comment (metakuni) [inline .happy/terraform/modules/sfn/main.tf:213]:
```suggestion
        "JobName": "seurat",
```

File: .happy/terraform/modules/sfn/main.tf [hunk 3]
```diff
@@ -195,6 +195,69 @@ resource "aws_sfn_state_machine" "state_machine" {
 EOF
 }
 
+
+resource "aws_sfn_state_machine" "state_machine_seurat" {
+  name     = "dp-${var.deployment_stage}-${var.custom_stack_name}-seurat-sfn"
+  role_arn = var.role_arn
+
+  definition = <<EOF
```
Comment (metakuni) [inline .happy/terraform/modules/sfn/main.tf:203]: Should add a comment indicating this step function temporarily lives in 2 repos: infra and portal

File: .happy/terraform/modules/sfn/main.tf (hunk 1 above)
Comment (ebezzi) [reply .happy/terraform/modules/sfn/main.tf:229]: Yeah, gonna remove them.

Comment (metakuni) [reply .happy/terraform/modules/sfn/main.tf:229]: As noted in the [infra PR](https://github.com/chanzuckerberg/single-cell-infra/pull/461/files#r759431351), we should keep the retries.

**Review by metakuni — APPROVED** [review]

**Pull request merged**